uaclient/util.py

```diff
@@ -183,25 +188,39 @@ def subp(args, rcs=None, capture=False):
                   for x in args]
     if rcs is None:
         rcs = [0]
-    try:
-        proc = subprocess.Popen(
-            bytes_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        (out, err) = proc.communicate()
-    except OSError:
+    while True:
         try:
-            if capture:
-                logging.error('Failed running cmd: %s, rc: %s stderr: %s',
-                              ' '.join(args), proc.returncode, err)
-        except UnboundLocalError:
-            pass
-        raise ProcessExecutionError(cmd=' '.join(args))
-    if proc.returncode not in rcs:
-        if capture:
-            logging.error('Failed running cmd: %s, rc: %s stderr: %s',
-                          ' '.join(args), proc.returncode, err)
-        raise ProcessExecutionError(
-            cmd=' '.join(args), exit_code=proc.returncode, stdout=out,
-            stderr=err)
+            proc = subprocess.Popen(
+                bytes_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            (out, err) = proc.communicate()
+        except OSError:
+            try:
+                message = ('Failed running cmd: %s, rc: %s stderr: %s' %
+                           (' '.join(args), proc.returncode, err))
+            except UnboundLocalError:
+                message = 'Failed running cmd: %s' % ' '.join(args)
+            if not retry_sleeps:
```

Comment:
I don't have time to dig in to this further, because I'm about to run to a session, but we shouldn't have the retrying logic duplicated both here and below; can we do something to remove the repetition?

Reply:
good thought, I was wondering how to do this. Spent a bit more time and teased out this logic into a separate _raise_or_retry helper function.

Reply:
this has been superceded by a change where we use subp as a retrying wrapper around _subp